applications/ShapeOptimizationApplication/custom_utilities/mesh_controller_utilities.h

```diff
@@ -0,0 +1,270 @@
+// ==============================================================================
+//  KratosShapeOptimizationApplication
+//
+//  License:         BSD License
+//                   license: ShapeOptimizationApplication/license.txt
+//
+//  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+//
+// ==============================================================================
+
+#ifndef MESH_CONTROLLER_UTILITIES_H
+#define MESH_CONTROLLER_UTILITIES_H
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// ------------------------------------------------------------------------------
+// External includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "includes/define.h"
+#include "processes/process.h"
+#include "includes/node.h"
+#include "includes/element.h"
+#include "includes/model_part.h"
+#include "includes/kratos_flags.h"
+#include "shape_optimization_application.h"
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+
+*/
+
+class MeshControllerUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of MeshControllerUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(MeshControllerUtilities);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    MeshControllerUtilities( ModelPart& modelPart )
+        : mrModelPart( modelPart )
+    {
+    }
+
+    /// Destructor.
+    virtual ~MeshControllerUtilities()
+    {
+    }
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    // ==============================================================================
+    void UpdateMeshAccordingInputVariable( const Variable<array_1d<double,3>> &rInputVariable )
+    {
+        KRATOS_TRY;
+
+        for(auto & node_i: mrModelPart.Nodes())
```

Comment:
const

Reply:
will probably not work because of non const X0() ...